### ISSUE TITLE
Add integration test for shadow dom 

### ISSUE THREAD

**Issue by jpettitt** [ampproject/amphtml#20044]
We've had multiple issues with DOM inconsistencies when AMP code is imported through a shadow dom based mechanism (in Polymer <amp-viewer>), see list below.  We need some shadow dom integration tests.

Here some of the issues:
#19434 amp-experiment
#19694 amp-subscriptions-google
#12912 amp-ads
#12388 amp-img
#12151 amp-list
#10799 amp-access

**Comment by justforfun** [comment]
We a brand new issue of the same kind:
https://github.com/ampproject/amphtml/issues/20400

**Comment by jpettitt** [comment]
Is this on the roadmap? 

cc @rudygalfi

**Comment by rudygalfi** [comment]
@rsimha let us know your thoughts here.

**Comment by rsimha** [comment]
I think this can be covered by E2E tests now. @prateekbh Can you comment?

**Comment by prateekbh** [comment]
Yes, I had a conversation around the same and definitely looks like this can be done via e2e tests
